### PR TITLE
add raft client on top of bucketclient API

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Use this client to fetch logs from bucketd in a way consistent between MetaData and bucketfile.

### CODE REVIEW

**Comment by DavidPineauScality** [comment]
I was wary of the name `RaftClient`, but it seems it's only a way to keep the state of the RaftSessionID along with a bucketclient instance, right ? Then it's okay (though I'd have prefered another name such as `RaftLogConsumer` or something showing a difference between 'RAFT' and the usage of 'Raft logs'

File: lib/storage/metadata/bucketclient/RecordLogProxy.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    if (err.code === 404) {
+                        // no such raft session, log and ignore
+                        this.logger.warn(
+                            'This raft session does not exist (yet)',
+                            { raftId: this.raftSession });
+                        return cbOnce(null, { info: { start: null,
+                                                      end: null } });
+                    }
+                    if (err.code === 416) {
+                        // requested range not satisfiable
```
Comment (rahulreddy) [inline lib/storage/metadata/bucketclient/RecordLogProxy.js]: Let's have a trace log here.

File: lib/storage/metadata/bucketclient/RecordLogProxy.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    _transform(itemObj, encoding, callback) {
+        itemObj.entries.forEach(entry => {
+            // eslint-disable-next-line no-param-reassign
+            entry.type = entry.type || 'put';
+        });
+        this.logger.debug('processing log item', { itemObj });
+        this.push(itemObj);
+        callback();
```
Comment (jonathan-gramain) [inline lib/storage/metadata/bucketclient/RecordLogProxy.js]: This should be ok, it's called by the node engine internally and the doc examples just call the callback directly as a way to notify that we processed the items.

**Comment by jonathan-gramain** [comment]
@ironman-machine try

**Pull request merged**